cmd/dc4bc_d/main.go

```diff
@@ -133,136 +152,63 @@ func genKeyPairCommand() *cobra.Command {
+				log.Fatalln("failed to prepare config: ", err)
 			}
 
-			kafkaTrustStorePath := viper.GetString(flagKafkaTrustStorePath)
-			kafkaTimeout := viper.GetDuration(flagKafkaTimeout)
-			tlsConfig, err := kafka_storage.GetTLSConfig(kafkaTrustStorePath)
-			if err != nil {
-				return fmt.Errorf("faile to create tls config: %w", err)
-			}
-
-			producerCredentials := viper.GetString(flagKafkaProducerCredentials)
-			producerCreds, err := parseKafkaSaslPlain(producerCredentials)
-			if err != nil {
-				return fmt.Errorf("failed to parse kafka credentials: %w", err)
-			}
-
-			consumerCredentials := viper.GetString(flagKafkaConsumerCredentials)
-			consumerCreds, err := parseKafkaSaslPlain(consumerCredentials)
-			if err != nil {
-				return fmt.Errorf("failed to parse kafka credentials: %w", err)
-			}
+			ctx := context.Background()
+			ctx, cancel := context.WithCancel(ctx)
 
-			storageDBDSN := viper.GetString(flagStorageDBDSN)
-			stg, err := kafka_storage.NewKafkaStorage(storageDBDSN, storageTopic, kafkaConsumerGroup, tlsConfig,
-				producerCreds, consumerCreds, kafkaTimeout)
+			sp, err := services.CreateServiceProviderWithCfg(cfg)
 			if err != nil {
-				return fmt.Errorf("failed to init storage client: %w", err)
+				log.Fatalf("failed to init service provider: %+v", err)
 			}
 
-			msgsToIgnore := viper.GetString(flagStorageIgnoreMessages)
-			useOffsetInsteadId := viper.GetBool(flagOffsetsToIgnoreMessages)
-			ignoredMsgs, err := parseMessagesToIgnore(msgsToIgnore, useOffsetInsteadId)
+			cli, err := node.NewNode(ctx, cfg, sp)
 			if err != nil {
-				return fmt.Errorf("failed to ignore messages in storage: %w", err)
-			}
-			if err := stg.IgnoreMessages(ignoredMsgs, useOffsetInsteadId); err != nil {
-				return fmt.Errorf("failed to ignore messages in storage: %w", err)
+				log.Fatalf("failed to init node: %+v", err)
 			}
 
-			keyStoreDBDSN := viper.GetString(flagStoreDBDSN)
-			keyStore, err := client.NewLevelDBKeyStore(username, keyStoreDBDSN)
-			if err != nil {
-				return fmt.Errorf("failed to init key store: %w", err)
-			}
-
-			framesDelay := viper.GetInt(flagFramesDelay)
-			chunkSize := viper.GetInt(flagChunkSize)
-
-			processor := qr.NewCameraProcessor()
-			processor.SetDelay(framesDelay)
-			processor.SetChunkSize(chunkSize)
-
-			cli, err := client.NewClient(ctx, username, state, stg, keyStore, processor)
-			if err != nil {
-				return fmt.Errorf("failed to init client: %w", err)
-			}
 			cli.SetSkipCommKeysVerification(viper.GetBool(flagSkipCommKeysVerification))
 
 			sigs := make(chan os.Signal, 1)
 			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
 			go func() {
 				<-sigs
 
-				log.Println("Received signal, stopping client...")
+				log.Println("Received signal, stopping node...")
 				cancel()
 
 				log.Println("BaseClient stopped, exiting")
 				os.Exit(0)
 			}()
 
-			listenAddress := viper.GetString(flagListenAddr)
+			server := http_api.RESTApiProvider{}
```

Comment:
This is kind of weird. http_api should just have a `New()` constructor; initialising an empty struct and then calling a method that sets things up is not idiomatic.

Reply:
done